### ISSUE TITLE
Failure to build with Xcode 11

### ISSUE THREAD

**Issue by fxcoudert** [halo/macosvpn#41]
```
==> xcodebuild SYMROOT=build
Build settings from command line:
    SYMROOT = build

note: Using new build system
note: Planning build
note: Constructing build description
error: SWIFT_VERSION '3.0' is unsupported, supported versions are: 4.0, 4.2, 5.0. (in target 'macosvpn' from project 'macosvpn')
```

**Comment by fxcoudert** [comment]
This is preventing users from installing as part of Homebrew on Mojave and Catalina

**Comment by halo** [comment]
Hi, thank you for your report. I'm painfully aware of it 😅 

When [this feature request](https://github.com/halo/macosvpn/issues/38) came in, I realized that much had to be done to get this project in shape for Xcode 10 and 11.

The good news is, that I'm 80% along the way to a pure Swift 5 implementation. The new version I'll be releasing within a matter of weeks is going to be compatible with macOS 10.10 and upwards.

I've never really taken the time to dig into how homebrew formulae specify compatibility for different operating system versions. I'd appreciate if someone could tell me what I'd have to do on my end to specify that. Naturally, there will be a changelog and a git tag for every release, but I'm not sure where I would specify which release matches which OS, so that installing with homebrew "simply works".

Actually, I just realize. Supporting 10.10 upwards will be totally sufficient (that's back to Yosemite from 2014, nobody should be on Mavericks any more!). So, everything is fine. I just need to invest some more time.

**Comment by fxcoudert** [comment]
> Supporting 10.10 upwards will be totally sufficient

Indeed.

**Comment by vldgeorgiev** [comment]
Hi @halo 
A few comments above you wrote that you are finishing a swift 5 version, and are planning to release it soon.
May I ask how close you are to that? 🙂
I have been reusing and building your older version of the code for a while and it works perfectly, but I wouldn't mind using the new one with Xcode 11.

**Comment by halo** [comment]
Good timing! Over the past month I could invest an hour or two per week to work on this. I deleted the last Objective-C file yesterday.

The conversion is basically complete, I pushed it to the branch called [swift](https://github.com/halo/macosvpn/tree/swift).

However, I need to increase spec coverage and right now, there is a segfault when I try to add a L2TP VPN service. More specifically, on [this line](https://github.com/halo/macosvpn/blob/bf6efe57491392e3c57f4762abf0feb81c7552df/macosvpn/Classes/ServiceConfigCreator.swift#L35).

So some more debugging is required, but I'm 99% along the way.

**Comment by vldgeorgiev** [comment]
Thank you for the information. And for the good work.
I will keep an eye on the repo for the upcoming changes.

**Comment by halo** [comment]
Good news. Specs [are passing](https://travis-ci.org/halo/macosvpn/builds/625102204). Now I'll just need to update the documentation, CHANGELOG, and fine-tune the `--help` output a little bit. I'll merge this in master soon.  🍰

**Comment by halo** [comment]
I went ahead and released version 1.0.0 officially. I'll close this for now, feel free to open a new issue should you encounter a bug.

Thank you for your patience and Merry Christmas 🎄
